components/apimgt/org.wso2.carbon.apimgt.impl/src/test/java/org/wso2/carbon/apimgt/impl/APIProviderImplTest.java

```diff
@@ -4359,4 +4363,34 @@ public void testAddFileToDocumentationForException() throws Exception {
         InputStream inputStream = Mockito.mock(InputStream.class);
         apiProvider.addFileToDocumentation(api.getId(), doc, fileName, inputStream, contentType);
     }
+
+    @Test
+    public void testSaveGraphqlSchemaDefinition() throws Exception {
+        Resource resource = new ResourceImpl();
+        String resourcePath = APIConstants.API_ROOT_LOCATION + RegistryConstants.PATH_SEPARATOR + "admin" + RegistryConstants.PATH_SEPARATOR +
+                "API1" + RegistryConstants.PATH_SEPARATOR + "1.0.0" + RegistryConstants.PATH_SEPARATOR;
+        String schemaContent = "sample schema";
+        APIIdentifier apiId = new APIIdentifier("admin", "API1", "1.0.0");
+        API api = new API(apiId);
+        Mockito.when(APIUtil.getGraphqlDefinitionFilePath("API1", "1.0.0", "admin")).thenReturn(resourcePath);
+
+        Resource resourceMock = Mockito.mock(Resource.class);
+        resourceMock.setContent(schemaContent);
+        resourceMock.setMediaType(String.valueOf(ContentType.TEXT_PLAIN));
+
+        //org.wso2.carbon.registry.api.RegistryException
```

Comment:
why is an exception name mentioned as a comment here?

Reply:
Added it to test registry exception

Reply:
Shall we mention it in the comment?